### PR TITLE
OAuth tutorial first draft

### PR DESCRIPTION
**Pull request by Neamar**

### CODE REVIEW

File: guides/tutorials/oauth.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+title: Creating a OAuth application
+subtitle: OAuth2 flow on AnyFetch
+layout: doc
+---
+
+AnyFetch implements OAuth2 flow.
+OAuth lets you request token on behalf of a user, and use the associated AnyFetch account according to the `scopes` you required (reading documents, writing documents...)
```
Comment (Boudedi) [inline guides/tutorials/oauth.md]: request tokenS

File: guides/tutorials/oauth.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+* **Application name**: the name of your Application. It will be used when the user is prompted to grant authorization on his account
+* **Description**: A few lines of description
+* **Callback URI**: a URL where the user will be transported back after giving consent. You may use the special value `urn:ietf:wg:oauth:2.0:oob` for out-of-browser flow (see specifications for details).
+* **OAuth application**: leave this unchecked if you're making a provider, else keep it on!
+* **Required scopes**: check all the scopes required by your application. The [endpoint list](/endpoints) indicates the required scope for each endpoint, just hover the lock icon.
+
+You will be able to edit those values later. For now, click on OK.
+
+You should see you application `app_id` and `app_secret`, keep the last one somewhere safe and private!
```
Comment (Boudedi) [inline guides/tutorials/oauth.md]: youR

File: guides/tutorials/oauth.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+* **replace_existing_token**: when the user is accessing your application for the first time, this parameter is ignored. Else, set this value to `true` to remove the previous token, skip the grant part and redirect automatically with a new code to your `redirect_uri`.
+* **approval_prompt**: when used with `replace_existing_token`, set the value to `true` to re-display the grant. For security reasons, if you updated your application scope, the grant page will always be displayed.
+
+The user will be prompted to authenticate, and should then see the following grant page:
+
+![Grant page](/images/tutorials/oauth/grant.png)
+
+When the user clicks "Allow", he will be redirected to your `redirect_uri` with a few querystring parameters:
+
+* **code**: an oauth code, to be traded later, the most important parameter
```
Comment (Boudedi) [inline guides/tutorials/oauth.md]: parameter.

File: guides/tutorials/oauth.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+* **return_to** the URL where the initial application required to be sent back once the OAuth flow complete. This is mostly used for providers, and can safely be ignored for OAuth applications.
```
Comment (Boudedi) [inline guides/tutorials/oauth.md]: completeS / is complete.

File: guides/tutorials/oauth.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+### Exchange code for access_token
+You should send a HTTP POST request to `https://manager.anyfetch.com/oauth/access_token`, with the following parameters:
+
+* **client_id**: your `app_id`, created on the first section
+* **client_secret**: your `app_secret`, created on the first section
+* **code**: thoe `code` sent to your `redirect_uri`. Be careful, the `code` is only valid for a few minutes.
```
Comment (Boudedi) [inline guides/tutorials/oauth.md]: thoe -> the

File: guides/tutorials/oauth.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+![App credentials](/images/tutorials/oauth/app_credentials.png)
+
+## Requesting a new token for a user
+Now that you have a new application, you probably want users.
+
+### Grant access
+To authorize a user, you need to redirect him to `https://manager.anyfetch.com/oauth/authorize?client_id={CLIENT_ID}&redirect_uri={ENCODED_REDIRECT_URI}`.
+
+> Redirect URI must match exactly the value you entered earlier.
```
Comment (saveman71) [inline guides/tutorials/oauth.md]: Not sure of exactly

File: guides/tutorials/oauth.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+Apart from that, the OAuth flow follows the [RFC6749 specification](https://tools.ietf.org/html/rfc6749).
+
+## Creating an application
+From your AnyFetch account, visit the "[Developers](https://manager.anyfetch.com/developers)" page on the manager website, and click on "[Create a new application](https://manager.anyfetch.com/developers/new)".
+
+You need to fill-in some values:
+
+* **Application name**: the name of your Application. It will be used when the user is prompted to grant authorization on his account
+* **Description**: A few lines of description
+* **Callback URI**: a URL where the user will be transported back after giving consent. You may use the special value `urn:ietf:wg:oauth:2.0:oob` for out-of-browser flow (see specifications for details).
```
Comment (saveman71) [inline guides/tutorials/oauth.md]: link ?

**Pull request merged**